msal/exceptions.py

```diff
@@ -27,12 +27,18 @@
 
 class MsalError(Exception):
     # Define the template in Unicode to accommodate possible Unicode variables
-    msg = u'An unspecified error'
+    msg = u'An unspecified error'  # Keeping for backward compatibility
 
-    def __init__(self, *args, **kwargs):
-        super(MsalError, self).__init__(self.msg.format(**kwargs), *args)
-        self.kwargs = kwargs
 
 class MsalServiceError(MsalError):
-    msg = u"{error}: {error_description}"
+    msg = u"{error}: {error_description}"  # Keeping for backward compatibility
+    def __init__(
+        self,
+        *args,
+        error: str, error_description: str,  # Historically required, keeping them for now
```

Comment:
Since it's only being kept "for now", is there any plan/backlog item to remove this at some point? Or is it something that might be needed again in the future?

Reply:
Good question. Perhaps I shall add more inline comments to elaborate:

1. We can't simply remove it, or else it will be a breaking change
2. We may, however, change required parameters into optional, at any time, without breaking anyone. However, such a change will be a one-way change, because once being optional, we will never be able to change it (back) to be required.
3. Since it was required, so we just keep it "required" for now, just in case that we would use it again.